### ISSUE TITLE
bug: Lifecycle events never be fired on React.FC

### ISSUE THREAD

**Issue by pjc0247** [ionic-team/ionic#20071]
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
I'm trying to receive lifecycle events following your documentation:
https://ionicframework.com/docs/react/lifecycle

https://github.com/pjc0247/bugreport/blob/master/src/pages/Home.tsx#L6

**Expected behavior:**
console log will be printed.

**Steps to reproduce:**
https://pjc0247.github.io/bugreport/build/index.html

**Related code:**
https://github.com/pjc0247/bugreport/blob/master/src/pages/Home.tsx#L6

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.4.12 (C:\Users\pjc02\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework : @ionic/react 4.11.6

Utility:

   cordova-res : 0.6.0
   native-run  : 0.2.8

System:

   NodeJS : v10.16.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.9.0
   OS     : Windows 10

```

**Comment by obedm503** [comment]
@pjc0247 beat me to it

This was introduced in ionic 4.11.6, version 4.11.5 works perfectly fine.
Here's another repo to reproduce the issue by comparing the 2 versions https://github.com/obedm503/ionic-hooks-broken

From what I've noticed this is only an issue on the first render, lifecycle hooks still work when navigating to other pages and back.

**Comment by elylucas** [comment]
Thanks,

I've found the issue and am working on a fix. Hopefully will be out later today.

**Comment by elylucas** [comment]
fixed and released in 4.11.7, thanks for reporting!

**Comment by obedm503** [comment]
can confirm it's working now
